swap/src/config.rs

```diff
@@ -1,110 +1,146 @@
+use crate::fs::ensure_directory_exists;
```

Comment:
This diff is unreadable. If you strongly feel that the swap CLI parameters to access the node must be named `config` and the blockchain characteristics and protocol parameters must be named `settings` I would recommend the following approach:

- introduce new module `configuration` in this PR for the node urls.
- Possible rename the `config` module to `settings`
- Do a follow-up PR where you rename `configuration` to `config`.

The criss-cross rename should be avoided for the sake of the reviewers.

Regarding the terminology for the blockchain times and expiries, the terminology is tricky and I believe it is because we mixed up some concepts. We are currently storing few types of information:
1. Blockchain characteristics/behaviour: avg block time, network
2. Security parameters: number of confirmation to finality, bob time to act
3. Protocol parameters: Bitcoin expiries

I think a bigger revamp to be done to separate those 3 concepts.

If you want a generic name I'd go for `parameters`.

Reply:
> This diff is unreadable. 

I split the changes into separate, isolated commits. Yes, the complete diff turned out to be ugly, but could you by any change review by commits?

Reply:
> If you strongly feel that the swap CLI parameters to access the node must be named config and the blockchain characteristics and protocol parameters must be named settings I would recommend the following approach.

I don't feel strong about it per-se, but it follows the idea of our other projects and is based on https://github.com/mehcode/config-rs as far as I could see.

Reply:
I tried to change the `settings` and `config` from the bottom up but it is quite a pain. This accounts to https://github.com/comit-network/xmr-btc-swap/pull/161#discussion_r566509040 as well. 

I think the current design is fine, and the commits are contained at the moment. Please review by commit if possible @D4nte - all commits should be contained now. 

As for the split between settings and config: I do think it is a good split because it emphasizes config (as in config-file) vs what is to be used within the program.
As for separating the wallet settings from the execution parameters: Those are related, because the wallets are needed for execution as well. Eventually we might move execution settings into the config as well. I also think that creating overall `testnet` settings is not wrong.

Would you be OK with merging this as is @D4nte or do you insist on changes?